presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -161,10 +152,15 @@ public PlanNodeDecorrelator(SymbolAllocator symbolAllocator, Lookup lookup)
         @Override
         public Optional<DecorrelationResult> visitLimit(LimitNode node, Void context)
         {
-            if (node.getCount() == 0 || node.isWithTies()) {
+            // do not bother to decorrelate a plan that should be pruned
+            if (node.getCount() == 0) {
                 return Optional.empty();
             }
 
+            if (node.isWithTies()) {
+                return visitPlan(node, null);
```

Comment:
why would execution even reach here if there are not correlations (implementation of `DecorrelatingVisitor#visitFilter` should prevent that).

Reply:
Does it matter how we get here? In case `node.isWithTies()` we are not doing any special decoration yet so calling `visitPlan(node, null);` conveys that.